File: app/src/main/java/in/smslite/fragments/ContactsPermissionFragment.java
package in.smslite.fragments;

import android.Manifest;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.v13.app.ActivityCompat;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.paolorotolo.appintro.ISlidePolicy;

import in.smslite.R;


public class ContactsPermissionFragment extends Fragment implements ISlidePolicy {
    private static final String TAG = ContactsPermissionFragment.class.getSimpleName();

    public ContactsPermissionFragment() {
        // Required empty public constructor
    }

    public static ContactsPermissionFragment newInstance() {
        return new ContactsPermissionFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_organize, container, false);
    }

    @Override
    public boolean isPolicyRespected() {
        return ContextCompat.checkSelfPermission(getContext(), Manifest.permission.READ_CONTACTS) == PackageManager.PERMISSION_GRANTED;
    }

    @Override
    public void onUserIllegallyRequestedNextPage() {
        Log.d(TAG, "Permission request");
        ActivityCompat.requestPermissions(getActivity(), new String[]{Manifest.permission.READ_CONTACTS}, 3);
    }
}

File: app/src/main/java/in/smslite/fragments/ContactsPermissionFragment.kt
package `in`.smslite.fragments

import `in`.smslite.R
import android.Manifest
import android.content.pm.PackageManager
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.legacy.app.ActivityCompat
import com.github.paolorotolo.appintro.ISlidePolicy

class ContactsPermissionFragment : Fragment(), ISlidePolicy {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_organize, container, false)
    }

    override fun isPolicyRespected(): Boolean {
        return ContextCompat.checkSelfPermission(context!!, Manifest.permission.READ_CONTACTS) == PackageManager.PERMISSION_GRANTED
    }

    override fun onUserIllegallyRequestedNextPage() {
        Log.d(TAG, "Permission request")
        ActivityCompat.requestPermissions(activity!!, arrayOf(Manifest.permission.READ_CONTACTS), 3)
    }

    companion object {
        private val TAG = ContactsPermissionFragment::class.java.simpleName
        fun newInstance(): ContactsPermissionFragment {
            return ContactsPermissionFragment()
        }
    }
}